packages/playground/src/playground-db.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace WordPress\Playground;
+
+defined('ABSPATH') || exit;
+
+/**
+ * Escape an array of values for use in a SQL query.
+ *
+ * @param array $array The array of values to escape.
+ * @return string The escaped values, separated by commas.
+ */
+function escape_array($array)
+{
+	global $wpdb;
+	$escaped = array();
+	foreach ($array as $value) {
+		if (is_numeric($value)) {
+			$escaped[] = $wpdb->prepare('%d', $value);
+		} else {
+			$escaped[] = $wpdb->prepare('%s', $value);
+		}
+	}
+	return implode(',', $escaped);
+}
+
+/**
+ * Create a database dump and add it to a zip archive.
+ *
+ * @param ZipArchive $zip The zip archive to add the database dump to.
+ */
+function dump_db($zip)
+{
+	global $wpdb;
+
+	$tables   = get_db_tables();
+	$sql_dump = array();
+
+	foreach ($tables as $table) {
+		array_push(
+			$sql_dump,
+			sprintf(
+				"DROP TABLE IF EXISTS %s;",
+				$wpdb->quote_identifier($table)
+			),
+			dump_db_schema($table)
+		);
+	}
+
+	foreach ($tables as $table) {
+		$records = $wpdb->get_results(
+			sprintf(
+				'SELECT * FROM %s',
+				$wpdb->quote_identifier($table)
+			),
+			ARRAY_A
+		);
+
+		if ($wpdb->last_error) {
+			error_log($wpdb->last_error);
+			continue;
+		}
+
+		foreach ($records as $record) {
+			array_push(
+				$sql_dump,
+				sprintf(
+					'INSERT INTO %1$s (%2$s) VALUES (%3$s);',
+					$wpdb->quote_identifier($table),
+					escape_array(
+						array_keys($record)
```

Comment:
Column names are identifiers, but they're escaped as values here. Should `$wpdb->quote_identifier` be used here, too?

Reply:
I used `$wpdb->prepare` here, but `$wpdb->quote_identifier` would be more appropriate here. 

Done in 1eb00ee177dc2371406681802d9c6802ce093a31

Reply:
Now when I think of it, we don't need column names or am I missing something?

This example works.

```
		array_push(
				$sql_dump,
				sprintf(
					'INSERT INTO %1$s VALUES (%2$s);',
					$wpdb->quote_identifier($table),
					escape_array(array_values($record))
				)
			);
```

Reply:
I removed column names in dd36deba4f2f20766e22fe023417e9e2ce46885e.